.github/workflows/full-matrix-tests.yml

```diff
@@ -41,83 +41,58 @@ on:
         - cron: "0 3 * * *"
 
 concurrency:
-    group: full-matrix-tests
-    cancel-in-progress: false
+    group: nightly-${{ github.head_ref || github.ref }}-${{ toJson(inputs) }}
+    cancel-in-progress: true
 
 # TODO matrix by workflow (`uses`) - not supported yet by GH
 jobs:
-    check-running-workflow:
-        runs-on: ubuntu-latest
-        outputs:
-            is_running: ${{ steps.check.outputs.is_running }}
-        steps:
-            - name: Check if the same workflow is running
-              id: check
-              uses: actions/github-script@v6
-              with:
-                  script: |
-                      const { data } = await github.rest.actions.listWorkflowRuns({
-                        owner: context.repo.owner,
-                        repo: context.repo.repo,
-                        workflow_id: context.workflow,
-                        status: 'in_progress',
-                      });
-                      const isRunning = data.workflow_runs.some(run => run.id !== context.runId);
-                      core.setOutput('is_running', isRunning.toString());
     check-input:
         runs-on: ubuntu-latest
-        needs: check-running-workflow
-        if: needs.check-running-workflow.outputs.is_running == 'false'
         steps:
-            - name: No tests selected
-              run: echo "No tests selected."
-              if: github.event_name == 'workflow_dispatch' && inputs.core == 'false' && inputs.java == 'false' && inputs.python == 'false' && inputs.node == 'false' && inputs.csharp == 'false' && inputs.go == 'false'
+            - name: no tests selected
+              run: false
+              if: github.event_name == 'workflow_dispatch' && inputs.core == false && inputs.java == false && inputs.python == false && inputs.node == false && inputs.java == false && inputs.csharp == false && inputs.go == false
 
     run-full-tests-for-core:
-        needs: check-running-workflow
-        if: needs.check-running-workflow.outputs.is_running == 'false' && inputs.core == 'true'
+        if: (github.repository_owner == 'valkey-io' && github.event_name == 'schedule') || (github.event_name == 'workflow_dispatch' && inputs.core == true)
         uses: ./.github/workflows/rust.yml
         name: Run CI for GLIDE core lib
         secrets: inherit
 
     # run-full-tests-for-redis-rs:
-    #     needs: check-running-workflow
-    #     if: needs.check-running-workflow.outputs.is_running == 'false' && inputs.redis-rs == 'true'
+    #     if: (github.repository_owner == 'valkey-io' && github.event_name == 'schedule') || (github.event_name == 'workflow_dispatch' && inputs.redis-rs == true)
```

Comment:
You can uncomment all, but change condition like we have for C# client.

Reply:
It fails on not finding the file.

Reply:
which file?

Reply:
`./.github/workflows/redis-rs.yml`
From line below, it doesn't exist in main yet.